tests/backup/backup_test.go

```diff
@@ -5484,6 +5483,231 @@ var _ = Describe("{ShareAndRemoveBackupLocation}", func() {
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step(fmt.Sprintf("Sharing full backup with view only access and incremental backup with full access with user [%s]", individualUser), func() {
+			log.InfoD("Sharing full backup [%s] with view only access and incremental backup [%s] with full access with user [%s]", fullBackupName, incrementalBackupName, individualUser)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = ShareBackup(fullBackupName, nil, []string{individualUser}, ViewOnlyAccess, ctx)
+			err = ShareBackup(incrementalBackupName, nil, []string{individualUser}, FullAccess, ctx)
+		})
+
+		Step("Validate that user with View Only access cannot restore or delete the backup", func() {
+			log.InfoD("Validate that user with View Only access cannot restore or delete the backup")
+
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(individualUser, "Password1")
+			log.FailOnError(err, "Fetching %s ctx", individualUser)
+			userContexts = append(userContexts, ctxNonAdmin)
+
+			// Register Source and Destination cluster
+			log.InfoD("Registering Source and Destination clusters from user context")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctxNonAdmin)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			// Start Restore and confirm that user cannot restore
+			restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+			err = CreateRestore(restoreName, fullBackupName, make(map[string]string), destinationClusterName, orgID, ctxNonAdmin)
+			log.Infof("Error returned - %s", err.Error())
+			// Restore validation to make sure that the user with View Access cannot restore
+			dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to restore backup"), true, "Verifying backup restore is not possible")
+
+			// Get Admin Context - needed to get backup UID
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			// Get Backup UID
+			backupDriver := Inst().Backup
+			backupUID, err := backupDriver.GetBackupUID(ctx, fullBackupName, orgID)
+			log.FailOnError(err, "Failed while trying to get backup UID for - %s", fullBackupName)
+
+			// Delete backup to confirm that the user cannot delete the backup
+			_, err = DeleteBackup(fullBackupName, backupUID, orgID, ctxNonAdmin)
+			dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
+		})
+
+		Step("Validate that user with View Only access on full backup and full access to incremental backup can restore", func() {
+			log.InfoD("Validate that user with View Only access on full backup and full access to incremental backup can restore")
+
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(individualUser, "Password1")
+			log.FailOnError(err, "Fetching %s ctx", individualUser)
```

Comment:
Can we move this outside step and inside it , so that we need not get the user context twice

Reply:
As we discussed, it is better to get the latest context because the token might have expired